mythril_core/src/acpi.rs

```diff
@@ -0,0 +1,201 @@
+#![deny(missing_docs)]
+use crate::error::{Error, Result};
+use byteorder::{ByteOrder, NativeEndian};
+use core::fmt;
+use core::ops::Range;
+
+/// Extented BIOS Data Area Start Address
+const EXTENDED_BIOS_DATA_START: usize = 0x000040e;
+/// Extented BIOS Data Area End Address
+const EXTENDED_BIOS_DATA_END: usize = EXTENDED_BIOS_DATA_START + 1024;
+/// Main BIOS Data Area Start Address
+const MAIN_BIOS_DATA_START: usize = 0x000e0000;
+/// Main BIOS Data Area End Address
+const MAIN_BIOS_DATA_END: usize = 0x000fffff;
+/// Well Known RSDP Signature
+const RSDP_SIGNATURE: &[u8; 8] = b"RSD PTR ";
+
+/// Offsets from [ACPI § 5.2.5.3]
+///
+/// [ACPI § 5.2.5.3]: https://uefi.org/sites/default/files/resources/ACPI_6_3_May16.pdf
+mod offsets {
+    use super::*;
+    /// Well known bytes, "RST PTR ".
+    pub const SIGNATURE: Range<usize> = 0..8;
+    /// Checksum of the fields defined by ACPI 1.0.
+    pub const CHECKSUM: usize = 8;
+    /// OEM-supplied ID string.
+    pub const OEMID: Range<usize> = 9..15;
+    /// Revision of the structure. Zero for ACPI 1.0 and two for ACPI 2.0.
+    pub const REVISION: usize = 15;
+    /// 32-bit physical address of the RSDT.
+    pub const RSDT_ADDR: Range<usize> = 16..20;
+    /// Length of the XSDT table in bytes (ACPI 2.0 only).
+    pub const LENGTH: Range<usize> = 20..24;
+    /// 64-bit physical address of the XSDT (ACPI 2.0 only).
+    pub const XSDT_ADDR: Range<usize> = 24..32;
+    /// Checksum of entire structure (ACPI 2.0 only).
+    pub const EXT_CHECKSUM: usize = 32;
+    /// Size of the RSDP, including the reservied field.
+    pub const SIZE: usize = 36;
```

Comment:
This should probably be outside of the `offsets` module, as it isn't really an offset (if I understand correctly).

Reply:
True. Would something like the following be reasonable?

```suggestion
    pub const RESERVED: Range<usize> = 33..36;
```

Then we would use `RESERVED.end` everywhere `SIZE` is used.

Reply:
Hmm, that seems a bit too magical to me. I think I'd rather a `RSDP_MAX_SIZE` defined outside the offests module. It seems like that what we're expressing here, right? The size is either 36 or 20 depending on version.

I suppose there is also the possibility that it is a 20 byte one right at the end, in which case we would miss it, right? Maybe we need a different approach.

Reply:
> I think I'd rather a RSDP_MAX_SIZE defined outside the offests module.

I created the `RSDP_V1_SIZE` and `RSDP_V2_SIZE` structures and then made the `search_range` function revision aware. So we could in theory find a V1 RSDP at the end of the slice.

Reply:
Awesome, I like that a lot more. Thanks